server/athenian/api/internal/miners/github/deployment.py

```diff
@@ -2674,31 +2667,33 @@ async def hide_outlier_first_deployments(
       the median of time distances of each deployment with the previous one
     """
     log = logging.getLogger(f"{__name__}.hide_outlier_first_deployments")
-    log.info("Searching for outlier first deployments")
+    log.info("searching for outlier first deployments")
     outlier_deploys = await _search_outlier_first_deployments(
         deployment_facts, meta_ids, mdb, log, threshold,
     )
-    computed_names = set(deployment_facts.index.values[computed_mask])
-    toclear_deploys = [d for d in outlier_deploys if d.deployment_name in computed_names]
-
-    if not toclear_deploys:
-        return
 
-    log.info("Clearing %d outlier first deployments", len(toclear_deploys))
-
-    stmts: list[Executable] = []
+    # group outlier deploys by name and consider only deploys still to be hidden
+    async def deploy_to_be_hidden(name: str, repositories: Sequence[str]) -> bool:
+        stmt = sa.select(1).where(
+            GitHubCommitDeployment.acc_id == account,
+            GitHubCommitDeployment.deployment_name == name,
+            GitHubCommitDeployment.repository_full_name.in_(repositories),
```

Comment:
```suggestion
```
better to remove this condition, will be more reliable

Reply:
Since you'll have to touch this, let's also union with similar checks in `GitHubPullRequestDeployment` and `GitHubReleaseDeployment`. It's impossible to have the latter without the former ofc, but who knows, maybe the process crashed somewhere on line 2704 where we clear those tables.